spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/HttpClientBeanPostProcessor.java

```diff
@@ -134,12 +135,38 @@ public void accept(HttpClientRequest req, Connection connection) {
 					return;
 				}
 			}
+
+			WrappedRequest request = new WrappedRequest(req);
+
+			// Check to see if there's a reference we need to use or update.
 			AtomicReference<Span> reference = req.currentContext()
-					.getOrDefault(AtomicReference.class, new AtomicReference<>());
-			WrappedHttpClientRequest request = new WrappedHttpClientRequest(req);
-			Span span = reference.get() == null ? handler().handleSend(request)
-					: handler().handleSend(request, reference.get());
-			reference.set(span);
+					.getOrDefault(AtomicReference.class, null);
+
+			Span span;
+			if (reference != null) {
+				span = reference.get() == null ? handler().handleSend(request)
+						: handler().handleSend(request, reference.get());
+				reference.set(span);
+			}
+			else { // there's no reference to update
+				span = handler().handleSend(request);
+			}
+
+			parseConnection(connection, span);
```

Comment:
@simonbasle I'm very curious about this instrumentation code.. there seem to be several entrypoints to start a span related to an http request and this here, I was unlucky as it isn't invoked in my example.

this one isn't either, and it has no reference to a connection.. https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceRequestHttpHeadersFilter.java#L86

this one is triggered, and it also has no reference to a connection... https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java#L166

1. Do you know how to trigger the line in this file? ex. `webClient.get().uri("/api").retrieve().bodyToMono(String.class);` does not.
2. for the other call sites, do you know how/when a connection can be parsed, if at all?

Until such time, it might be better to delete the connection parsing code client side, as I don't know if it works (eventhough the server side does)

Reply:
I killed this part of the PR so that it can be merged.. we can discuss the remote address thing client side offline or in another change.